icu4c/source/test/cintltst/cdattst.c

```diff
@@ -2017,4 +2019,23 @@ static void TestNarrowQuarters(void) {
     }
 }
 
+static void TestExtraneousCharacters(void) {
+    // regression test for ICU-21802
+    UErrorCode err = U_ZERO_ERROR;
+    UCalendar* cal = ucal_open(u"UTC", -1, "en_US", UCAL_GREGORIAN, &err);
+    UDateFormat* df = udat_open(UDAT_PATTERN, UDAT_PATTERN, "en_US", u"UTC", -1, u"yyyyMMdd", -1, &err);
+    udat_setLenient(df, false);
```

Comment:
Shouldn't this be inside the following block after the test for successful creation of the date formatter?

Reply:
You're right; it should.  Fixed.